dowhy/do_sampler.py

```diff
@@ -50,11 +62,11 @@ def __init__(
 
         """
         self._data = data.copy()
-        self._causal_model = causal_model
-        self._target_estimand = self._causal_model.identify_effect()
-        self._target_estimand.set_identifier_method("backdoor")
-        self._treatment_names = parse_state(self._causal_model._treatment)
-        self._outcome_names = parse_state(self._causal_model._outcome)
+        self._target_estimand = identify_effect_auto(
```

Comment:
any specific reason you removed the `set_identifier_method` call?

Reply:
I put it back in. I think my thought here was that the `identify_effect_auto` method should set the "identifier method", but I might mix something up here. Doesn't the `identify_effect_auto` select the identifier method (although it is currently by default the backdoor)?